### DISCUSSION TITLE
How do you calculate the analogWrite bit for a given freqency?

### DISCUSSION THREAD

**Discussion by drakejest** [earlephilhower/arduino-pico#1365] [Q&A]
I would like to know what is the highest number of bits for the analogWriteRes() i can do for a given frequency? Specifically 90kHz PWM frequency. What is the formula?

**Comment by earlephilhower** [comment]
https://github.com/earlephilhower/arduino-pico/blob/07781e9cf56e2ca9828726fc8de01f5bec170675/cores/rp2040/wiring_analog.cpp#L87-L103.  We obey the frequency first, then derate the bits so it can happen w/o needing a bit clock > fSYS.

Enable core debug and debug Serial output and it'll dump the calculated values.
